cmake/compiler_settings.cmake

```diff
@@ -67,10 +67,8 @@ else ()
 endif ()
 
 if (CMAKE_CXX_COMPILER_ID MATCHES "Clang")
-  # Disable this particular warning because `-isystem` is being to
-  # sent to clang when assembling, and is at that point unrecognized.
-  # TODO: See #760: Fix this when errors are better propagated.
-  add_compile_options(-Wno-error=unused-command-line-argument)
+  # While compiling ASM files, suppress unused command line argument errors.
+  add_compile_options($<$<COMPILE_LANGUAGE:ASM>:-Wno-error=unused-command-line-argument>)
```

Comment:
While this fixes the build, it does so just by ignoring the warning that we're sending unused command line arguments when assembling. I would certainly prefer fixing it such that we didn't cause the warning in the first place, but this can be a backup option.